### DISCUSSION TITLE
Nuxt 3 with external API keeping cookie

### DISCUSSION THREAD

**Discussion by danya-rnd** [nuxt/nuxt#16459] [General]
🔥🔥🔥 Hey there! 🔥🔥🔥

I had a big problem to pass requests both from server side and client side with new $fetch of nuxt3 with cookies.

For those who faced the same problem I will share my implementation. May be this is not a good practice, but it works 😁

First of all I'm using the [proxy](https://www.npmjs.com/package/@nuxtjs-alt/proxy) developed by @Teranode. Thanks a lot for him, he made a big job!

**Nuxt3 config:**
``` javascript
import { defineNuxtConfig } from 'nuxt'

export default defineNuxtConfig({
  runtimeConfig: {
    public: {
      baseAPI: process.env.BASE_API || 'http://localhost:4000',
    },
  },
  modules: ['@nuxtjs-alt/proxy'],
  proxy: {
    '/api/v1/': {
      target: process.env.BASE_API || 'http://localhost:4000',
      changeOrigin: true,
    },
  },
})
```

**The composable was used to return the baseURL if process is server.**
```javascript
export default function () {
  const config = useRuntimeConfig();
  return process.server && { baseURL: config.public.baseAPI };
}
```

**$fetch usage example**
```html
<script lang="ts" setup>
const { data } = await $fetch(`/api/v1/example`, {
  ...useBaseURL(),
  headers: useRequestHeaders(["cookie"]),
});
</script>
```

The secret was that server side request should be passed with full path. Client side requests works through a proxy.

I'm sharing my **[repo](https://github.com/danya-rnd/nuxt3-cookie-support)** for everyone who needs it.

Hope I helped someone! Good luck!⚡️

**Comment by AaronConlon** [comment]
@danya-rnd hello, where from `useBaseURL` function?

**Reply by danya-rnd** [reply]
Hello. This is the composable from composables folder in my repo.

**Comment by typhoon11** [comment]
Still this is giving an error
![image](https://user-images.githubusercontent.com/66431066/200188968-42154baf-6ef0-46c4-b1c2-bb9d239e68f3.png)

**Reply by danya-rnd** [reply]
Hi! Provide me your code
